src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.cs

```diff
@@ -265,6 +267,42 @@ internal abstract partial class AbstractAddImportCodeFixProvider<TSimpleNameSynt
         }
 
         /// <summary>
+        /// We ignore references that are in a directory that contains the names "Packages".
+        /// These directories are most likely the ones produced by NuGet, and we don't want
+        /// to offer to add .dll reference manually for dlls that are part of NuGet packages.
+        /// </summary>
+        private bool IsInPackagesDirectory(PortableExecutableReference reference)
+        {
+            const string Packages = nameof(Packages);
+
+            try
+            {
+                if (!string.IsNullOrWhiteSpace(reference.FilePath))
+                {
+                    if (reference.FilePath.IndexOf(Packages, StringComparison.OrdinalIgnoreCase) >= 0)
+                    {
+                        var currentPath = reference.FilePath;
+                        while (currentPath != null)
+                        {
+                            var name = Path.GetFileName(currentPath);
+                            if (StringComparer.OrdinalIgnoreCase.Equals(name, Packages))
+                            {
+                                return true;
+                            }
+
+                            currentPath = Path.GetDirectoryName(currentPath);
+                        }
+                    }
+                }
+            }
+            catch
```

Comment:
I really don't like this style of wrapping everything in a catch, it basically hides bugs. For example, if we found, theoretically, that Path.GetDirectoryName doesn't work for UNC paths, we would never find this out until a customer complained about it - because we would never crash. None of these methods should throw if your logic and reference.FilePath didn't have a bug.